devito/finite_differences/differentiable.py

```diff
@@ -86,7 +87,11 @@ def dimensions(self):
     @property
     def indices_ref(self):
         """The reference indices of the object (indices at first creation)."""
-        return EnrichedTuple(*self.dimensions, getters=self.dimensions)
+        if len(self._args_diff) == 1:
```

Comment:
I'm not sure I can discern between this `if` and the `elif` below. What does ``len(self._args_diff) == 1`` really mean, ie what case does it represent? and same for the elif below

Perhaps add a comment?

is it like, if == 1, then it's embedding a further Differentiable object, which therefore needs processing?

also, if == 0 is the base case, I suggest to swap the order of the ifs:

```
if len(self._args_diff) == 0:
    # Comment
    ...
elif len(self._args_diff) == 1:
    # Comment
    ....
else:
     # Comment
    ...
```

I'm also not sure why do == 2 , == 3...  need to be treated differently than == 1 ? Sorry, this is probably just me being slow...

Reply:
the len() == 1 is litteraly the linked issue